docs/source/reference/jinja.rst

```diff
@@ -77,7 +107,28 @@ Adds escape characters to JSON strings.
 
 .. code-block:: bash
 
-    {{value_key | json_escape}}
+    {{ value_key | json_escape }}
+
+jsonpath_query
+~~~~~~~~~~~~~~
+
+Provides the ability to extract data from complex ``object`` data using the
+`JSONPath <http://goessner.net/articles/JsonPath/>` query language. More specifically
+we use the ``jsonpath-rw`` library that has its own extensions, details can be
+found on the `jsonpath-rw GitHub page <https://github.com/kennknowles/python-jsonpath-rw/tree/master/jsonpath_rw>`.
+Data passed into this function should be of type ``object`` or ``array``.
+The result of this function will either be an array of results, or None if the
+query did not return any results.
+If you would like to test out your JSONPath queries prior to utilizing this filter
+an online evaluator can be found `here <http://jsonpath.com/>`.
+
+.. code-block:: bash
+
+    # input  = {'people': [{'first': 'James', 'last': 'd'},
+    #                      {'first': 'Jacob', 'last': 'e'},
+    #                      {'first': 'Jayden', 'last': 'f'}]}
+    # result = ['James', 'Jacob', 'Jayden']
+    {{ input | jmespath_query('people[*].first') }}
```

Comment:
I think for such complex filter we need several more examples to show people how powerful it is.

Reply:
Good idea! Added more examples.